### DISCUSSION TITLE
Semantic commits type always set to `chore`

### DISCUSSION THREAD

**Discussion by alexandernanberg** [renovatebot/renovate#10258] [Request Help]
**How are you running Renovate?**

- [x] WhiteSource Renovate hosted app on github.com
- [ ] Self hosted

**Describe the bug**

Renovate always uses `chore` type when creating PRs even though runtime/production dependencies have been updated. [Docs](https://docs.renovatebot.com/semantic-commits/) says 

> Renovate uses chore by default, but uses fix for updates to your production dependencies in your package.json file.

But that does not seem to be the case. 

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
NFO: Repository started
{
  "renovateVersion": "25.34.0"
}
DEBUG: Using localDir: /mnt/renovate/gh/acast-tech/roles-api
DEBUG: Repository cache is valid
DEBUG: initRepo("acast-tech/roles-api")
DEBUG: Overriding default GitHub endpoint
{
  "endpoint": "https://api.github.com/"
}
DEBUG: No dangling containers to remove
DEBUG: acast-tech/roles-api default branch = master
DEBUG: Using app token for git init
DEBUG: resetMemCache()
DEBUG: Resetting npmrc
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: Checking cached config file name
DEBUG: Existing config file confirmed
DEBUG: Repo is onboarded
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base",
      "group:allNonMajor",
      "schedule:monthly"
    ],
    "lockFileMaintenance": {
      "enabled": true,
      "schedule": [
        "before 3am on the first day of the month"
      ]
    },
    "npmrc": "***********",
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
DEBUG: migrated config
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base",
      "group:allNonMajor",
      "schedule:monthly"
    ],
    "lockFileMaintenance": {
      "enabled": true,
      "schedule": [
        "before 3am on the first day of the month"
      ]
    },
    "npmrc": "***********",
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
DEBUG: Found encrypted config
{
  "config": {
    "npmToken": "***********"
  }
}
DEBUG: Trying default padding for npmToken
DEBUG: Trying RSA_PKCS1_PADDING for npmToken
DEBUG: Decrypted npmToken
DEBUG: Migrating npmToken to npmrc
{
  "decryptedToken": "39************************************07"
}
DEBUG: Replacing ${NPM_TOKEN} with decrypted token
DEBUG: Found npmrc in decrypted config - setting
DEBUG: Setting npmrc
DEBUG: Ignoring any .npmrc files in repository due to configured npmrc
DEBUG: Setting hostRules from config
DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
DEBUG: detectSemanticCommits()
DEBUG: No vulnerability alerts found
DEBUG: No vulnerability alerts found
DEBUG: processRepo()
DEBUG: No baseBranches
DEBUG: extract()
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

N/A

**Comment by HonkingGoose** [comment]
This seems more of a configuration help issue.
There are ways to configure Renovate to use the commit style you like best for (dev) dependencies.

I'll let the Renovate maintainers help you with the config.

**Reply by rarkins** [reply]
There'salso  not enough info provided to be able to answer anything.

**Reply by HonkingGoose** [reply]
> There'salso not enough info provided to be able to answer anything.

I guessed that as well, but wasn't sure...

---

@alexandernanberg Read https://docs.renovatebot.com/semantic-commits/ to learn how to customize the commit messages from Renovate bot.

**Reply by alexandernanberg** [reply]
I've read it but maybe misunderstanding? The docs says

> Renovate uses chore by default, but uses fix for updates to your production dependencies in your package.json file.

Doesn't that mean that if `deps` in `dependencies` updates, `fix` will be used instead of `chore`?

**Reply by HonkingGoose** [reply]
> Doesn't that mean that if `deps` in `dependencies` updates, `fix` will be used instead of `chore`?

That should be the default behavior, I guess...

---

I think we need more information to be able to help you.

@rarkins Can you explain what we need from @alexandernanberg to be able to them out?

**Reply by rarkins** [reply]
That documentation is misleading. Renovate defaults `semanticCommitType` to `chore`: https://github.com/renovatebot/renovate/blob/f5ae2d5db17039698900d24fe0dc8fefd394fb67/lib/config/definitions.ts#L1154-L1159

If you extend `config:base` then you get this rule included: https://github.com/renovatebot/renovate/blob/f5ae2d5db17039698900d24fe0dc8fefd394fb67/lib/config/presets/internal/default.ts#L184-L197

**Reply by HonkingGoose** [reply]
PR incoming to fix the docs. 😄 

---

See PR #10269.

**Reply by alexandernanberg** [reply]
Alright, but if you look at the attached logs I'm extending `config:base` already. Not sure if `group:allNonMajor` or `schedule:monthly` interferes? 

```
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base",
      "group:allNonMajor",
      "schedule:monthly"
    ],
    "lockFileMaintenance": {
      "enabled": true,
      "schedule": [
        "before 3am on the first day of the month"
      ]
    },
    "npmrc": "***********",
    "encrypted": {
      "npmToken": "***********"
    }
  }
}
```

**Reply by rarkins** [reply]
We need a reproduction repo to look into it further
